src/driver-steelseries.c

```diff
@@ -597,50 +635,92 @@ create_cycle(struct steelseries_led_cycle *cycle)
 	list_init(&cycle->points);
 }
 
+static int
+create_cycle_spec(struct steelseries_led_cycle_spec *spec, struct ratbag_device *device)
+{
+	spec->has_2_led_ids = false; /* assume for now that the default is 1 ID */
+
+	switch (device->ids.product) {
+		case STEELSERIES_PRODUCT_ID_RIVAL310:
+		case STEELSERIES_PRODUCT_ID_SENSEI310:
+			spec->hid_report_type = HID_OUTPUT_REPORT;
+			spec->header_len = 28;
+			spec->cmd_val = STEELSERIES_ID_LED;
+			spec->led_id_idx = 2;
+			spec->duration_idx = 3;
+			spec->repeat_idx = 19;
+			spec->trigger_idx = 23;
+			spec->point_count_idx = 27;
+			return 0;
+		case STEELSERIES_PRODUCT_ID_RIVAL600:
+			spec->hid_report_type = HID_FEATURE_REPORT;
+			spec->header_len = 30;
+			spec->cmd_val = STEELSERIES_ID_LED_INTENSITY_SHORT;
+			spec->has_2_led_ids = true;
+			spec->led_id_idx = 2;
+			spec->led_id2_idx = 7;
+			spec->duration_idx = 8;
+			spec->repeat_idx = 24;
+			spec->trigger_idx = 25;
+			spec->point_count_idx = 29;
+			return 0;
+
+		default:
+			return -1;
+	}
+}
```

Comment:
@phomes isn't there a better way to do this? I feel this should be based on the protocol version, not the device IDs.

Reply:
Yes, that does make more sense